src/profile-logic/profile-data.js

```diff
@@ -253,6 +253,23 @@ export function getTimingsForPath(
   };
   let rootTime = 0;
 
+  function accumulateToTimings(
+    timings: {
+      breakdownByImplementation: BreakdownByImplementation | null,
+      value: number,
+    },
+    stackIndex: IndexIntoStackTable,
+    funcIndex: IndexIntoFuncTable
+  ): void {
+    timings.value += interval;
+
+    let implementation = 'native';
+    if (funcTable.isJS[funcIndex]) {
+      implementation = getJsImplementationForStack(stackIndex, thread);
+    }
+    accumulateDataToBreakdown(timings, implementation);
```

Comment:
Ok, after spending a bit more time with the code, I think `accumulateToTimings` and `accumulateDataToBreakdown` should be combined into one function, unless I'm missing something I see no reason why it needs to be two. I found the two functions confusing to read. Also, phrasing the function as `accumulateDataToTimings` seems to answer the question of what we are actually accumulating to timings.

```js
/**
   * This is a small utility function to more easily add data to the timing
   * and breakdowns.
   */
  function accumulateDataToTimings(
    timings: {
      breakdownByImplementation: BreakdownByImplementation | null,
      value: number
    },
    stackIndex: IndexIntoStackTable,
    funcIndex: IndexIntoFuncTable
  ): void {
    timings.value += interval;

    const implementation = funcTable.isJS[funcIndex]
      ? getJsImplementationForStack(stackIndex, thread)
      : "native";

    if (timings.breakdownByImplementation === null) {
      timings.breakdownByImplementation = {};
    }
    if (timings.breakdownByImplementation[implementation] === undefined) {
      timings.breakdownByImplementation[implementation] = 0;
    }
    timings.breakdownByImplementation[implementation] += interval;
  }
```

Reply:
yeah that's right; I believe this stems from the fact that I extracted `accumulateToTimings` further from some code that already had `accumulateDataToBreakdown`. Naming was hard especially because the purpose of the 2 functions wasn't clear, which is another indication they should be merged. Thanks !